### PR TITLE
User Can Read Email Feature

### PR DESCRIPTION
**Pull request by sealfury**
#### On This Branch We:
- Create a user f-bot
- Add Selenium and Webdriver gems (currently unused)
- Add Warden helpers
- Write a simple precursor test to check whether f-bot can log in
- Write a test to check whether a user can read an email in their inbox

### CODE REVIEW

File: spec/features/user_can_read_email_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+require "rails_helper"
+require "pry"
```
Comment (tochman) [inline spec/features/user_can_read_email_spec.rb]: I don't think you need to require `pry` here. It is required by default by the rails helper

File: spec/features/user_can_read_email_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+
```
Comment (emtalen) [inline spec/features/user_can_read_email_spec.rb]: remove extra line

File: spec/features/user_can_read_email_spec.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    it 'display the inbox button' do
+        expect(page).to have_content 'Inbox'
+    end
+
+    it 'navigate to inbox' do
+        click_on 'Inbox'
+        expect(page).to have_content 'Sent'
+        expect(page).to have_content 'Trash'
+    end
+end
```
Comment (emtalen) [inline spec/features/user_can_read_email_spec.rb]: This test is a little bit 'kaka på kaka', you are in a way testing this functionality when reading the email.

File: spec/features/user_can_read_email_spec.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
+require "rails_helper"
```
Comment (tochman) [inline spec/features/user_can_read_email_spec.rb]: You don't have to require this file in every test file. It is required by default using the setting in `.rspec` in the project folder

File: spec/features/user_can_read_email_spec.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    
+    it 'can read a message' do
+        click_on "View"
+        expect(page).to have_text 'A message'
+    end
+
+    # it 'compose' do
+    #     click_on "Compose"
+    #     expect(page).to have_field "Text"
+    # end
```
Comment (emtalen) [inline spec/features/user_can_read_email_spec.rb]: remove commented out code

File: spec/features/user_can_read_email_spec.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    
+end
```
Comment (tochman) [inline spec/features/user_can_read_email_spec.rb:20]: Please remote commented code. Also, indent your code using 2 spaces.

Comment (emtalen) [inline spec/features/user_can_read_email_spec.rb:20]: remove a lot of extra lines

File: spec/features/user_can_read_email_spec.rb (hunk 3 above)
Comment (sealfury) [reply spec/features/user_can_read_email_spec.rb]: Thanks- we realised this test was superfluous but left it in so you could see our thought process in writing a sort of f-bot sanity check test.

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        it { is_expected.to validate_presence_of :email }
+        it { is_expected.to validate_presence_of :password }
+    end
+
+    describe 'Factory' do
+        it 'expects a valid Factory' do
+            expect(FactoryBot.create(:user)).to be_valid
+        end
+    end
+end
```
Comment (emtalen) [inline spec/models/user_spec.rb:18]: Just because you made it looks so nice with 2 spaces indentation in the other file, please fix it here also 😉

**Pull request merged**